src/Microsoft.FeatureManagement/FeatureFilters/TimeWindowFilter.cs

```diff
@@ -56,7 +57,19 @@ public Task<bool> EvaluateAsync(FeatureFilterEvaluationContext context)
                 return Task.FromResult(false);
             }
 
-            return Task.FromResult((!settings.Start.HasValue || now >= settings.Start.Value) && (!settings.End.HasValue || now < settings.End.Value));
+            //
+            // Hit the first occurrence of the time window
+            if ((!settings.Start.HasValue || now >= settings.Start.Value) && (!settings.End.HasValue || now < settings.End.Value))
+            {
+                return Task.FromResult(true);
+            }
+
+            if (settings.Recurrence != null)
+            {
```

Comment:
So the recurrence evaluation gets called on every request. I'd like for us to consider the implications of that. For example, is any caching needed?

Reply:
The validation of recurrence settings also gets called on every request. I can move the validation to `BindParameters`.

To implement a cache for recurrence evaluator, we should at least make some change to the FeatureFilterEvaluationContext. The reason is that: One feature flag can have multiple time window filters. We need the filter index information to distinguish different time window filters of the same feature flag so that we can retrieve the correct cache key "featurename_timewindowfilter_index_recurrence_cache"